x11-libs/libxaw/libxaw-1.0.13.recipe

```diff
@@ -0,0 +1,117 @@
+SUMMARY="X Athena Widget library"
+DESCRIPTION="X Athena Widgets, or Xaw, is a widget set based on the X Toolkit \
+Intrinsics (Xt) Library. Xaw & Xt form together what is known as the X Toolkit."
+HOMEPAGE="https://www.x.org/releases/individual/lib/"
+COPYRIGHT="1985-1990, 1994, 1998 The Open Group
+	2003-2004 Roland Mainz
+	1998-1999 The XFree86 Project, Inc.
+	1991 OMRON Corporation
+	1985-1998 Digital Equipment Corporation
+	1989 Prentice Hall
+	1985-1989, 1991, 1994 X Consortium"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://www.x.org/releases/individual/lib/libXaw-$portVersion.tar.bz2"
+CHECKSUM_SHA256="8ef8067312571292ccc2bbe94c41109dcf022ea5a4ec71656a83d8cce9edb0cd"
+SOURCE_DIR="libXaw-$portVersion"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libXaw6Version="6.0.1"
+libXaw7Version="7.0.0"
+
+libXaw6VersionCompat="$libXaw6Version compat >= ${libXaw6Version%%.*}"
+libXaw7VersionCompat="$libXaw7Version compat >= ${libXaw7Version%%.*}"
+
+PROVIDES="
+	libxaw$secondaryArchSuffix = $portVersion
+	lib:libxaw6$secondaryArchSuffix = $libXaw6VersionCompat
+	lib:libxaw7$secondaryArchSuffix = $libXaw7VersionCompat
```

Comment:
Don't worry, this fits with tabspace = 4, which is also [applied to][0] HaikuPorts

```diff
-	lib:libxaw6$secondaryArchSuffix = $libXaw6VersionCompat
-	lib:libxaw7$secondaryArchSuffix = $libXaw7VersionCompat
+	lib:libxaw${libVersionCompat##* }$secondaryArchSuffix = $libVersionCompat
+	lib:libxaw${libVersionCompat2##* }$secondaryArchSuffix = $libVersionCompat2
```

[0]: https://github.com/haikuports/haikuports/pull/2010#discussion_r159844382

Reply:
> BTW, I think we need a shorthand for `${libVersionCompat##* }`. Do you have any suggestions for the name?

Maybe we could choose `libMajorVersion`. What do you think of that?

But for this recipe I think it would be better to stick to hardcoded major versions, since `libXaw.so.8` is not part of the source tarball and is almost not used by any X client.
And if any update ever happens, it will probably be for libXaw7 only.

Reply:
`libMajorVersion` sounds good enough for me.

However, I think hardcoding also works here. HaikuPorter would catch when `lib:libxaw8` came up from nowhere. And hopefully, we would catch cases where someone forgot to bump the numbers in `INSTALL()`

Reply:
:)

For the second part of your comment, I'm not sure I understand correctly. Do you think we should use define and use `libMajorVersion` in this recipe?
I'd say that, for this recipe, it would not be a good idea to add this variable because we know that there libXaw8 will probably never be shipped in this tarball. And if this ever happens, we would just need to change one digit.
We don't really need more flexibility than what we already have with `libVersion6` and `libVersion7`. Don't you agree?

Reply:
I've looked into Debian changelog. Looks like there [was][0] Xaw8, but upstream dropped it, so I don't think it's much of a concern any more. So I agree with `libVersion6` and `libVersion7`

[0]: https://tracker.debian.org/news/138754